### PR TITLE
Fix for accepting spaces in arguments, at instantiation of Audioset.

### PR DESCRIPTION
**Pull request by somsubhojit1988**

### CODE REVIEW

**Comment by lukefahr** [comment]
Can you make this into 4 different pull requests?  I want to make some tweaks to 1 or 2 of them.  

-A

**Comment by somsubhojit1988** [comment]
Actually they are 4 different commits. I had pushed them incrementally, onto my master and made pull request from my master to your's which is the  reason why they are being shown as consolidated. I will make 4 different branches and make pull requests for each of them separately.

File: audioset.py [hunk 1]
```diff
@@ -41,7 +41,7 @@ class AudioSetBuilder (object):
     def __init__(this, audioset='balanced',
                 eval_file = None, balanced_file = None, unbalanced_file = None, \
                 ontol_file = None,\
-                data_dir=None, logLvl=logging.INFO):
+                 data_dir=None, logLvl=logging.INFO, sampling_rate=None):
```
Comment (lukefahr) [inline audioset.py:44]: Can you move sampling_rate to it's own line above data_dir.

Comment (lukefahr) [reply audioset.py:44]: And give it a default value?

File: audioset.py [hunk 2]
```diff
@@ -62,11 +62,8 @@ def __init__(this, audioset='balanced',
                         unbalanced_file=unbalanced_file, ontol_file=ontol_file,
                         logLvl=logLvl)
         this.cdl = ClipDownloader( data_dir = this.ddir , 
-                        logLvl=logLvl) 
+                                   logLvl=logLvl, sampling_rate=sampling_rate)
```
Comment (lukefahr) [inline audioset.py:65]: again, sampling_rate before logLvl

File: audioset.py [hunk 3]
```diff
@@ -62,11 +62,8 @@ def __init__(this, audioset='balanced',
                         unbalanced_file=unbalanced_file, ontol_file=ontol_file,
                         logLvl=logLvl)
         this.cdl = ClipDownloader( data_dir = this.ddir , 
-                        logLvl=logLvl) 
+                                   logLvl=logLvl, sampling_rate=sampling_rate) 
 
-
-    #
-    #
```
Comment (lukefahr) [inline audioset.py:69]: Don't delete the '#' lines

File: audioset.py [hunk 4]
```diff
@@ -576,7 +574,7 @@ def _download_clip(this, ytid, data_dir, start=0.0, stop=None):
         this.log.info('Downloading: ' + str(ytid) )
         this.log.debug('\t\t' + ' into ' + str(ddir))
 
-        fname = ddir + '/' + ytid + '.wav'
+        fname = ddir + '/'+ ytid + '.wav'
```
Comment (lukefahr) [inline audioset.py:577]: don't delete space

File: audioset.py [hunk 5]
```diff
@@ -461,14 +458,15 @@ def _openAudSet(this, fname):
 class ClipDownloader(object):
     ''' downloader for youtube clips '''
     
-    def __init__(this, data_dir=None, logLvl= logging.WARN):
+    def __init__(this, data_dir=None, sampling_rate = None,logLvl= logging.WARN):
```
Comment (lukefahr) [inline audioset.py:461]: We might want to rethink this a little better.  I'm worried you are going to end up with a bunch of downsampled clips in the same data directory as non-downsampled clips, and audioset won't know which are which.  Perhaps we want to put in a warning or something if the pre-downloaded clip is a sampling rate less than what you have requested?  Or re-download it?

File: audioset.py [hunk 6]
```diff
@@ -461,14 +458,15 @@ def _openAudSet(this, fname):
...
 
         this.log = logging.getLogger( type(this).__name__)
         this.log.setLevel(logLvl)
         this.log.info ("Created")
 
         this.mydir = os.path.dirname(os.path.realpath(__file__))
         this.ddir =  data_dir
+        this.sampling_rate = sampling_rate
```
Comment (lukefahr) [inline audioset.py:469]: move above this.mydir

File: audioset.py [hunk 7]
```diff
@@ -652,11 +651,20 @@ def _crop_clip(this, in_fname, out_fname, start=0, stop=None):
         ffmpeg_args += ['-y']
         ffmpeg_args += ['-i', in_fname ]
         ffmpeg_args += ['-ss', str(start)]
-
+
```
Comment (lukefahr) [inline audioset.py:654]: remove spaces?

**Review by lukefahr — CHANGES_REQUESTED** [review]

**Comment by lukefahr** [comment]
Commit 1 is merged.  I split Commit 2 into 2 different commits and merged.  Commit 3 needs some work.  Commit 4 will merge once 3 is fixed.  Please make a seperate branch for Commit 3, and another one for Commit 4.  

Thanks

**Pull request closed without merging**